sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -196,6 +196,14 @@ object SQLConf {
     .booleanConf
     .createWithDefault(true)
 
+  val NO_UNESCAPED_SQL_STRING = buildConf("spark.sql.noUnescapedStringLiteral")
+    .internal()
+    .doc("Since Spark 2.0, we use unescaped SQL string for string literals including regex. " +
+      "It is different than 1.6 behavior. Enabling this config can use no unescaped SQL string " +
+      "literals and mitigate migration problem.")
```

Comment:
How about
> When true, string literals (including regex patterns) remains escaped in our SQL parser. The default is false since Spark 2.0. Setting it to `true` can restore the behavior prior to Spark 2.0.